shared_model/interfaces/common_objects/range_types.hpp

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_SHARED_MODEL_RANGE_TYPES_HPP
+#define IROHA_SHARED_MODEL_RANGE_TYPES_HPP
+
+#include <boost/range/any_range.hpp>
+
+namespace shared_model {
+  namespace interface {
+
+    class Signature;
+    class Transaction;
+    class AccountAsset;
+
+    namespace types {
+
+      /// Type of signature range, which returns when signatures are invoked
+      using SignatureRangeType = boost::any_range<interface::Signature,
+                                                  boost::forward_traversal_tag,
```

Comment:
Why is it forward traversal?

Reply:
https://en.cppreference.com/w/cpp/container/unordered_set

> iterator | Constant ForwardIterator

I think it is because `unordered_set` has forward iterator, and this range is used in signable.